editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/BeanConfigUtil.java

```diff
@@ -0,0 +1,525 @@
+		IMavenProjectFacade facade = MavenPlugin.getMavenProjectRegistry().create(project, new NullProgressMonitor());
+		if(facade != null){
+			IPath[] paths = facade.getCompileSourceLocations();
+			if (paths != null && paths.length > 0) {
+				for (IPath p : paths) {
+					if (p == null)
+						continue;
+					IResource res = project.findMember(p);
+					if (res != null) {
+						return javaProject.getPackageFragmentRoot(res);
+					}
+				}
+			}
+		}
+		return null;
+	}
+	
+	private IPackageFragmentRoot findPackageFragmentRoot(final IProject project, IJavaProject javaProject) throws CoreException {
+		IPackageFragmentRoot fromFacade = findPackageFragmentRootWithFacade(project, javaProject);
+		if (fromFacade != null) {
+			return fromFacade;
+		} else {
+			IPackageFragmentRoot[] allPackageFragmentRoots = javaProject.getAllPackageFragmentRoots();
+			if(allPackageFragmentRoots.length == 1) {
+				return allPackageFragmentRoots[0];
+			} else {
+				IFolder tstFolder = project.getFolder("src/main/java"); //$NON-NLS-1$
+	            IPackageFragmentRoot tstRoot = javaProject.getPackageFragmentRoot(tstFolder);
+	            if (tstRoot.exists()) {
+	            	return tstRoot;
+	            } else {
+	            	tstFolder.create(false, true, null);
+	            }
+				IClasspathEntry[] entries = javaProject.getRawClasspath();
+				IClasspathEntry[] newEntries = new IClasspathEntry[entries.length + 1];
+				System.arraycopy(entries, 0, newEntries, 0, entries.length);
+				newEntries[entries.length] = JavaCore.newSourceEntry(tstRoot.getPath());
+				javaProject.setRawClasspath(newEntries, new NullProgressMonitor());
+				return tstRoot;
+			}
+		}
+	}
+	
+	public String handleClassBrowse(IProject project, Shell shell) {
+		IJavaSearchScope scope = null;
+		if (project != null) {
+			IJavaProject jproject = JavaCore.create(project);
+			if (jproject == null) {
+				scope = SearchEngine.createWorkspaceScope();
+			} else {
+				scope = SearchEngine.createJavaSearchScope(new IJavaElement[] { jproject });
+			}
+		}
+
+		try {
+			SelectionDialog dialog = JavaUI.createTypeDialog(shell, null, scope,
+					IJavaElementSearchConstants.CONSIDER_CLASSES, false, "*Bean"); //$NON-NLS-1$
+			if (dialog.open() == SelectionDialog.OK) {
+				Object[] result = dialog.getResult();
+				if (result.length > 0 && result[0] instanceof IType) {
+					return ((IType) result[0]).getFullyQualifiedName();
+				}
+			}
+		} catch (JavaModelException e) {
+			CamelEditorUIActivator.pluginLog().logError(e);
+		}
+		return null;
+	}
+
+	private String openNoArgMethodDialog(IJavaProject jproject, String className, Shell shell) throws JavaModelException {
+		IType foundClass = jproject.findType(className);
+		if (foundClass != null) {
+			IMethod[] methods = foundClass.getMethods();
+			ArrayList<IMethod> noParamsList = new ArrayList<>();
+			for (int i = 0; i < methods.length; i++) {
```

Comment:
code can be simplified using streams, see https://github.com/bfitzpat/fuseide/pull/5